### PR TITLE
Global clipboard events - copy and paste

### LINKED ISSUE #149: Enable pasting SVG without clicking the "Paste markup" button
In Figma, if you have SVG code in your clipboard and just hit Ctrl+V (or Cmd+V on macOS) in a document, it creates a new frame with your SVG in that document. It's a really nice way to work with SVG quickly.

It would be great to have the same in SVGOMG. It probably needs no UI. Only need to listen for the right keyboard shortcut on the page. Though I guess there might be security restrictions that Figma works around with an invisible input, I’m not sure.

### PR DESCRIPTION
**Pull request by gismya**
Added global copy and paste events. Ctrl+c copies the current SVG markup (Like the "Copy to text-button") and ctrl+v now globally works like the paste markup-box.

Fixes #149

### CODE REVIEW

**Comment by XhmikosR** [comment]
There should at least be some kind of visual indication IMHO. Maybe use a Toast?

File: src/js/page/main-controller.js
Comment (jakearchibald) [inline src/js/page/main-controller.js:134]:
```suggestion
    const val = event.clipboardData.getData('text').trim();    
```

File: src/js/page/main-controller.js
Comment (jakearchibald) [inline src/js/page/main-controller.js:144]:
```suggestion
```

The toast is great for copy, but I don't think it's needed for paste. The UI already makes it clear that something happened.

File: src/js/page/ui/copy-button.js
Comment (jakearchibald) [inline src/js/page/ui/copy-button.js:29]: Could you rename this to indicate it performs an action? Right now it reads like it "gets the text to be copied". Maybe `copyText`?

**Review by jakearchibald — CHANGES_REQUESTED** [review]
Almost there! Nice feature

File: src/js/page/ui/copy-button.js
Comment (XhmikosR) [inline src/js/page/ui/copy-button.js:50]: Nitpicking these 2 newlines are not needed :)

File: src/js/page/ui/main-menu.js
Comment (XhmikosR) [inline src/js/page/ui/main-menu.js:62]: Not sure about using `Event`; `nanoevents` are used everywhere else.

Comment (jakearchibald) [reply src/js/page/ui/main-menu.js:62]: It's a little bit of a hack, but if an event is going to be used here, it's the right kind of event, since this is an `HTMLElement`.

**Comment by jakearchibald** [comment]
@TrySound @XhmikosR I'm happy to merge this as-is and create a seperate ticket for a bit of refactoring. Or do you think that should be done before merging?

Feels a bit weird having the controller call into the copy button when keys are pressed. Instead, the copy functionality should be moved into the controller or a utility, and the controller calls it when the button is pressed, or when the keys are pressed.

Similarly with paste, right now it's calling into the menu to set the paste input and manufacturing an input event. Seems like this should be moved a level up.

**Comment by XhmikosR** [comment]
The only thing I think it makes sense to have before this lands is a consistent usage of `copySupported`.

The rest I leave it to you guys :)

File: src/js/page/ui/copy-button.js
Comment (gismya) [reply src/js/page/ui/copy-button.js:50]: This is why people invented automatic linting ;)

**Comment by jakearchibald** [comment]
I don't think we need `copySupported` anymore, since all modern browsers support it. @gismya do you mind removing the usage of that as part of this PR? No worries if not, I'm happy to do it.

**Review by TrySound — APPROVED** [review]

**Comment by gismya** [comment]
@jakearchibald - I agree that going into the UI components to access the functions felt extremely hacky - but didn't want to come in and refactor too much "uninvited". I feel an utility would be the most suitable place. I can look into making those changes tonight probably.

File: .vscode/launch.json
Comment (TrySound) [inline .vscode/launch.json:1]: I would not commit vscode config here

**Comment by gismya** [comment]
Haha, I'm too slow with reading all the updated. I'll remove copySupported

File: .vscode/launch.json
Comment (gismya) [reply .vscode/launch.json:1]: Fudge - mistake from my part. Managed to remember to remove it in every commit until now. My bad!

I'll add the .vscode-folder to gitignore :)

Comment (TrySound) [reply .vscode/launch.json:1]: Maybe ignore it globally?

Comment (gismya) [reply .vscode/launch.json:1]: Did not know I could have a global gitignore. Now that's set up :)

**Comment by gismya** [comment]
Didn't meditate too long on the refactor, so I apologize in advance for any breaches of code conventions, but I made another PR for my proposed refactoring: https://github.com/gismya/svgomg/pull/1

**Review by jakearchibald — APPROVED** [review]
This is great, thanks so much for adding this!

**Pull request merged**
